File: ruby/BetaPowerEfficientSpaceStation.rb

module Deepspace
  class BetaPowerEfficientSpaceStation < PowerEfficientSpaceStation
    @@EXTRAEFFICIENCY = 1.20

    def initialize(station)
      super(station)
      @dice=Dice.new
    end

    def fire
      fire = 0
      if(@dice.extraEfficiency)
        fire = super()*@@EXTRAEFFICIENCY
      else
        fire = super()
      end
      return fire
    end

    def getUIversion
      return BetaPowerEfficientSpaceStationToUI.new(self)
    end

    def to_s
      return getUIversion.to_s
    end
  end
end

File: PRACTICA4/deepspace/src/deepspace/BetaPowerEfficientSpaceStation.java
package deepspace;

public class BetaPowerEfficientSpaceStation extends PowerEfficientSpaceStation{
    private static final float EXTRAEFFICIENCY= 1.2F;
    private Dice dice;
    public BetaPowerEfficientSpaceStation(SpaceStation station){
        super(station);
        dice=new Dice();
    }

    @Override
    public float fire(){
        float f;
        if(dice.extraEfficiency()) f=super.fire()*EXTRAEFFICIENCY;
        else f=super.fire();

        return f;
    }

    @Override
    public String toString(){
        String salida = "[Beta]" + super.toString();
        return salida;
    }

    @Override
    public BetaPowerEfficientSpaceStationToUI getUIversion(){
        return new BetaPowerEfficientSpaceStationToUI(this);
    }

}